kernel/rtlil.cc

```diff
@@ -280,6 +280,30 @@ int RTLIL::Const::as_int(bool is_signed) const
 	return ret;
 }
 
+void RTLIL::Const::compress(bool is_signed)
+{
+	if (bits.empty()) return;
+
+	// back to front (MSB to LSB)
+	RTLIL::State leading_bit;
+	if(is_signed)
+		leading_bit = (bits.back() == RTLIL::State::Sx) ? RTLIL::State::S0 : bits.back();
+	else
+		leading_bit = RTLIL::State::S0;
+
+    size_t idx = bits.size();
+    while (idx > 0 && bits[idx -1] == leading_bit) {
+        --idx;
+    }
+
+	// signed needs one leading bit
+	if (is_signed && idx < bits.size()) {
```

Comment:
I think this check needs to be `idx + 1 < bits.size()`, take e.g. `4'bxxxx`

Reply:
Looking at it with fresh eyes the check looks ok, nevermind

Reply:
In the case of `4'bxxxx`:  
Starts with `idx=4`, the bit `bits[4-1]` does not match the leading bit (which is never `X`).  
So we have `idx=4` which means `idx < bits.size()` is false and then it erases the vector from end to end which should do nothing.

But the case with unsigned all zeros is actually a problem because it will create a `idx=0` at the end which would delete the entire vector. -> fixed